build-system/eslint-rules/html-template.js

```diff
@@ -159,32 +147,146 @@ module.exports = function (context) {
     return matches;
   }
 
-  return {
-    CallExpression(node) {
-      if (/test-/.test(context.getFilename())) {
-        return;
-      }
+  function cannotImplicitlyReturnCalls(node) {
+    context.report({
+      node,
+      message: `Do not implicitly return ${node.tag.callee.name}()\`...\`, since it requires a helper constant in function scope.`,
+      fix: function* (fixer) {
+        yield fixer.insertTextBefore(node, `{\nreturn (`);
+        yield fixer.insertTextAfter(node, `);\n}`);
+      },
+    });
+  }
 
-      const {callee} = node;
-      if (callee.type !== 'Identifier') {
-        return;
-      }
+  const checkedScopeBlocks = new Set();
+  function tagMustBeHelperConstant(node) {
+    const {name} = node.tag.callee;
+    // Fixer inserts constant when required:
+    //   const html = htmlFor(...);
+    const helperConstName = staticTemplateFactories[name];
+    context.report({
+      node,
+      message: `Do not use ${name}() call directly as template tag. Store its result in a helper constant named "${helperConstName}", and use that as the template tag. This enforces code format, and enables syntax highlighting in some editors.`,
+      fix: function* (fixer) {
```

Comment:
I think this fixer is doing a bit too much effort for a linter. We need to consider that the identifiers are from new bindings, etc.

Reply:
> identifiers are from new bindings, etc.

Do you mean that we could potentially override `html` or it might reference a different value?

This was not a problem applying the fixer on the entire codebase, except for a test of `htmlFor` itself.